main.go

```diff
@@ -272,43 +271,16 @@ func setupScheduler(mgr ctrl.Manager, cCache *cache.Cache, queues *queue.Manager
 	}
 }
 
-func blockForPodsReady(cfg *config.Configuration) bool {
+func blockForPodsReady(cfg *configapi.Configuration) bool {
 	return waitForPodsReady(cfg) && cfg.WaitForPodsReady.BlockAdmission != nil && *cfg.WaitForPodsReady.BlockAdmission
 }
 
-func waitForPodsReady(cfg *config.Configuration) bool {
+func waitForPodsReady(cfg *configapi.Configuration) bool {
 	return cfg.WaitForPodsReady != nil && cfg.WaitForPodsReady.Enable
 }
 
-func encodeConfig(cfg *config.Configuration) (string, error) {
-	codecs := serializer.NewCodecFactory(scheme)
-	const mediaType = runtime.ContentTypeYAML
-	info, ok := runtime.SerializerInfoForMediaType(codecs.SupportedMediaTypes(), mediaType)
-	if !ok {
-		return "", fmt.Errorf("unable to locate encoder -- %q is not a supported media type", mediaType)
-	}
-
-	encoder := codecs.EncoderForVersion(info.Serializer, config.GroupVersion)
-	buf := new(bytes.Buffer)
-	if err := encoder.Encode(cfg, buf); err != nil {
-		return "", err
-	}
-	return buf.String(), nil
-}
-
-func apply(configFile string) (ctrl.Options, config.Configuration, error) {
-	var err error
-	options := ctrl.Options{
-		Scheme: scheme,
-	}
-	cfg := config.Configuration{}
-
-	if configFile == "" {
-		scheme.Default(&cfg)
-		options, err = options.AndFrom(&cfg)
-	} else {
-		options, err = options.AndFrom(ctrl.ConfigFile().AtPath(configFile).OfKind(&cfg))
-	}
+func apply(configFile string) (ctrl.Options, configapi.Configuration, error) {
+	options, cfg, err := config.Load(scheme, configFile)
 	if err != nil {
 		return options, cfg, err
 	}
```

Comment:
Can we add a TODO so that the code below goes to pkg/config?

Maybe `config.Load` can receive the set of available integrations.

Reply:
I'd prefer not to,  we need to import `jobframwork` and some of the jobs so the integration list is not empty. This is main reason I did not move this part already.